### PR TITLE
feat: Algorithm for Rotating linked list 'N' times

### PR DESCRIPTION
**Pull request by nikgautamgithub**
Algorithm for rotating a linked list to the right any number of times.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 1]
```diff
@@ -0,0 +1,210 @@
+/**
+ * @file
+ * @brief Rotate a linked list 'n' number of times.
+ * @details
+ * The linked list is rotated to right side by one step 'n' times.
+ **/
+
+#include <cassert> /// for assert
+#include "bits/stdc++.h"
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:9]: Please do not use `bits/stdc++.h` as it includes lots of other functions and unused variables that slow the compilation process.
It's best to include the headers/libraries one by one separately to ensure you're only loading headers/libraries you will need or use. 🙂

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: operations_on_datastructures/linked_list_right_rotation.cpp (hunk 1 above)
Comment (nikgautamgithub) [reply operations_on_datastructures/linked_list_right_rotation.cpp]: Done.

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+ * @brief Rotate a linked list 'n' number of times.
+ * @details
+ * The linked list is rotated to right side by one step 'n' times.
+ **/
+
+#include <cassert> /// for assert
+#include <iostream>
+#include <memory>
+
+namespace operations_on_datastructures
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:12]:
```suggestion
/**
 * @namespace operations_on_datastructures
 * @brief Operations on Data Structures
 */
namespace operations_on_datastructures
```

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        std::shared_ptr<Node> t = head; ///< pointer to iterate through linked list
+        while (t->next != nullptr)
+        {
+            std::cout << t->next->data << " ";
+            t = t->next;
+        }
+        std::cout << "\n";
+    }
+
+} ///< operation_on_datastructures
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:177]:
```suggestion
}  // namespace operation_on_datastructures
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Nice work! 🚀
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/nikgautamgithub/C-Plus-Plus/actions

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 4]
```diff
@@ -0,0 +1,215 @@
+/**
+ * @file
+ * @brief Rotate a linked list 'n' number of times.
+ * @details
+ * The linked list is rotated to right side by one step 'n' times.
+ **/
+
+#include <cassert> /// for assert
+#include <iostream> /// for IO operations
+#include <memory> /// for dynamic memory allocator
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:10]:
```suggestion
#include <cassert>  /// for assert
#include <iostream> /// for IO operations
#include <memory>   /// for dynamic memory allocator
```

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 5]
```diff
@@ -0,0 +1,215 @@
+/**
+ * @file
+ * @brief Rotate a linked list 'n' number of times.
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:3]: Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+ * @details
+ * The linked list is rotated to right side by one step 'n' times.
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:5]: Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..

File: operations_on_datastructures/linked_list_right_rotation.cpp [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+}  // namespace operation_on_datastructures
+
+/**
+ * @brief Main function
+ * @param argc commandline argument count (ignored)
+ * @param argv commandline array of arguments (ignored)
+ * @returns 0 on exit
+ */
+
```
Comment (Panquesito7) [inline operations_on_datastructures/linked_list_right_rotation.cpp:189]: Please add self-test cases so we can verify your program works properly (**see the [typical structure of a program](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) as a reference**). Let us know if you need any help here or on our [**Discord server**](https://the-algorithms.com/discord). 😃

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Good work! 🚀

**Pull request closed without merging**